pkg/daemon/daemon.go

```diff
@@ -240,6 +245,54 @@ func NewClusterDrivenDaemon(
 	return dn, nil
 }
 
+const (
+	logindNewSessionMessageIDMatch = "MESSAGE_ID=8d45620c1a4348dbb17410da57c60c66"
+)
+
+func (dn *Daemon) detectEarlySSHAccesses() error {
+	journal, err := sdjournal.NewJournal()
+	if err != nil {
+		return err
+	}
+	defer journal.Close()
+	if err := journal.AddMatch(logindNewSessionMessageIDMatch); err != nil {
+		return err
+	}
+	if err := journal.SeekHead(); err != nil {
+		return err
+	}
+	var detected bool
+	for {
+		r, err := journal.Next()
+		if err != nil {
+			return err
+		}
+		// if the journal ended
+		if r == 0 {
+			break
+		}
+		entry, err := journal.GetEntry()
+		if err != nil {
+			return err
+		}
+		// this is so fragile, but from testing, it's pretty accurate as a way
+		// to detect if there were sessions created before we took over with the daemon
+		if entry != nil && strings.Contains(entry.Fields["MESSAGE"], "New session") {
```

Comment:
We don't even need to check the `MESSAGE` here. The msg id already represents the event we're looking for.

Reply:
Also, there's no need for a for loop here, right? We can just `.Next()` and then `.GetEntry()` sequentially.

Reply:
for the second point, just 1 entry is already enough to mark the node ssh/accessed so yeah, no need for a loop for sequentially calling `GetEntry`